src/msm.rs

```diff
@@ -74,89 +74,86 @@ impl VariableBaseMSM {
+                is_neg_normal,
+            );
         });
 
         bucket_msm.process_complete();
-        return bucket_msm.batch_reduce();
+        bucket_msm.batch_reduce()
     }
 
     fn multi_scalar_mul_general<P: Parameters>(
         points: &[GroupAffine<P>],
         scalars: &[BigInt<P>],
         window_bits: u32,
         max_batch: u32,
-        max_collisions: u32
+        max_collisions: u32,
     ) -> GroupProjective<P> {
         let scalar_size = <P::ScalarField as PrimeField>::size_in_bits() as u32;
         let num_slices: u32 = (scalar_size + window_bits - 1) / window_bits;
-        let mut bucket_msm = BucketMSM::<P>::new(
-            scalar_size,
-            window_bits,
-            max_batch,
-            max_collisions,
-        );
+        let mut bucket_msm =
+            BucketMSM::<P>::new(scalar_size, window_bits, max_batch, max_collisions);
         let mut slices: Vec<u32> = vec![0; num_slices as usize];
 
-        let scalars_and_bases_iter = scalars
-            .iter()
-            .zip(points)
-            .filter(|(s, _)| !s.is_zero());
+        let scalars_and_bases_iter = scalars.iter().zip(points).filter(|(s, _)| !s.is_zero());
         scalars_and_bases_iter.for_each(|(&scalar, point)| {
             Self::msm_slice::<P>(scalar, &mut slices, window_bits);
-            bucket_msm.process_point_and_slices(&point, &slices);
+            bucket_msm.process_point_and_slices(point, &slices);
         });
 
         bucket_msm.process_complete();
-        return bucket_msm.batch_reduce();
+        bucket_msm.batch_reduce()
     }
 
     pub fn multi_scalar_mul_custom<P: Parameters>(
         points: &[GroupAffine<P>],
         scalars: &[BigInt<P>],
         window_bits: u32,
         max_batch: u32,
-        max_collisions: u32
+        max_collisions: u32,
     ) -> GroupProjective<P> {
-        assert!(window_bits as usize > GROUP_SIZE_IN_BITS,
-                "Window_bits must be greater than the default log(group size)");
+        assert!(
+            window_bits as usize > GROUP_SIZE_IN_BITS,
+            "Window_bits must be greater than the default log(group size)"
+        );
         if TypeId::of::<P>() == TypeId::of::<G1Parameters>() {
             Self::multi_scalar_mul_g1_glv(points, scalars, window_bits, max_batch, max_collisions)
         } else {
             Self::multi_scalar_mul_general(points, scalars, window_bits, max_batch, max_collisions)
         }
     }
 
-    pub fn multi_scalar_mul<P: Parameters>(points: &[GroupAffine<P>], scalars: &[BigInt<P>]) -> GroupProjective<P> {
+    pub fn multi_scalar_mul<P: Parameters>(
+        points: &[GroupAffine<P>],
+        scalars: &[BigInt<P>],
+    ) -> GroupProjective<P> {
         let opt_window_size = Self::get_opt_window_size(log2(points.len()));
-        Self::multi_scalar_mul_custom(&points, &scalars, opt_window_size, 2048, 256)
+        Self::multi_scalar_mul_custom(points, scalars, opt_window_size, 2048, 256)
```

Comment:
Why should the "&" be removed?

Reply:
Since `points` & `scalaras` is already a reference, in fact, `&&` value was passed inside, which is pointless

Reply:
Makes sense